services/src/main/java/org/keycloak/protocol/oid4vc/issuance/mappers/OID4VCIssuedAtTimeClaimMapper.java

```diff
@@ -0,0 +1,141 @@
+ */
+package org.keycloak.protocol.oid4vc.issuance.mappers;
+
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.protocol.ProtocolMapper;
+import org.keycloak.protocol.oid4vc.model.CredentialSubject;
+import org.keycloak.protocol.oid4vc.model.VerifiableCredential;
+import org.keycloak.provider.ProviderConfigProperty;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Map issuance date to the credential, under the default claim name "iat"
+ * <p>
+ * subjectProperty can be used to change the claim name.
+ * <p>
+ * Source of the information can either be computed, or read from the VerifiableCredential object
+ * bearing other claims. Default is the value in the verifiable credential.
+ * <p>
+ * We will use the java.time.temporal.ChronoUnit enum values to help flatten down the time.
+ *
+ * @author <a href="mailto:francis.pouatcha@adorsys.com">Francis Pouatcha</a>
+ */
+public class OID4VCIssuedAtTimeClaimMapper extends OID4VCMapper {
+
+    public static final String MAPPER_ID = "oid4vc-issued-at-time-claim-mapper";
+
+    // Omit value if defaults to "iat"
+    public static final String SUBJECT_PROPERTY_CONFIG_KEY = "subjectProperty";
+
+    // We will use the java.time.temporal.ChronoUnit enum values to help flatten down the time.
+    // Omit property if no truncation.
+    public static final String TRUNCATE_TO_TIME_UNIT_KEY = "truncateToTimeUnit";
+
+    // Time computed (COMPUTE) or taken from the verifiable credential (VC).
+    // Defaults to VC. Falls back to COMPUTE.
+    public static final String VALUE_SOURCE = "valueSource";
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES = new ArrayList<>();
+
+    static {
+        ProviderConfigProperty subjectPropertyNameConfig = new ProviderConfigProperty();
+        subjectPropertyNameConfig.setName(SUBJECT_PROPERTY_CONFIG_KEY);
+        subjectPropertyNameConfig.setLabel("Issuance Date Claim Property Name");
+        subjectPropertyNameConfig.setHelpText("Name of the property to contain the iat.");
+        subjectPropertyNameConfig.setType(ProviderConfigProperty.STRING_TYPE);
+        CONFIG_PROPERTIES.add(subjectPropertyNameConfig);
+
+        ProviderConfigProperty truncateToTimeUnit = new ProviderConfigProperty();
+        truncateToTimeUnit.setName(TRUNCATE_TO_TIME_UNIT_KEY);
+        truncateToTimeUnit.setLabel("Truncate To Time Unit");
+        truncateToTimeUnit.setHelpText("Truncate to time unit");
+        truncateToTimeUnit.setType(ProviderConfigProperty.STRING_TYPE);
+        CONFIG_PROPERTIES.add(truncateToTimeUnit);
+
+        ProviderConfigProperty valueSource = new ProviderConfigProperty();
+        valueSource.setName(VALUE_SOURCE);
+        valueSource.setLabel("Value Source");
+        valueSource.setHelpText("Value Source");
```

Comment:
From the help text, it is not clear what is the point of this configuration option (unless you look at the code, which typical user won't do). 

Is it perhaps possible to rename this and change the type to BOOLEAN_TYPE, so that it is displayed as On/Off flag in the admin console UI? Also is it possible to update the help-text to be more clear to describe what means value ON and what means value OFF? If I understand correctly, the ON could mean `Instant.now()` when the OFF could mean `Optional.ofNullable(verifiableCredential.getIssuanceDate())` (or vice-versa, whatever makes more sense).

Reply:
Yes. But if ```Optional.ofNullable(verifiableCredential.getIssuanceDate())``` is empty, we still fallback to ```Instant.now()```

Went for LIST_TYPE, as we might have more options in the future. WDYT?

modified with [77c10ee](https://github.com/keycloak/keycloak/pull/31620/commits/77c10ee12b66aae61c096da1248cb2e356ed7d6b)

Reply:
@francis-pouatcha Yes, list is good if you think that more flexibility is needed in the future. So +1 for this idea.

Added one very small suggestion to the help text to clearify a bit what the value `vc` means (sent other comment, which is displayed in this PR as a "suggestion").